polynote-frontend/polynote/ui/component/notebook/cell.ts

```diff
@@ -73,35 +81,35 @@ export class CellContainer extends Disposable {
     readonly el: TagElement<"div">;
     private readonly cellId: string;
     private cell: Cell;
+    private cellState: StateHandler<CellState>;
 
-    constructor(private dispatcher: NotebookMessageDispatcher, private notebookState: NotebookStateHandler, private cellState: StateHandler<CellState>, private path: string) {
+    constructor(private dispatcher: NotebookMessageDispatcher, private notebookState: NotebookStateHandler, state: StateHandler<CellState>, private path: string) {
         super()
+        const cellState = this.cellState = state.fork(this);
         this.cellId = `Cell${cellState.state.id}`;
         this.cell = this.cellFor(cellState.state.language);
         this.el = div(['cell-component'], [this.cell.el]);
         this.el.click(evt => this.cell.doSelect());
-        cellState.view("language").addObserver((newLang, oldLang) => {
+        cellState.view("language").addObserver((newLang, update) => {
             // Need to create a whole new cell if the language switches between code and text
-            if (oldLang !== undefined && (oldLang === "text" || newLang === "text")) {
+            if (update.oldValue && (update.oldValue === "text" || newLang === "text")) {
                 const newCell = this.cellFor(newLang)
                 newCell.replace(this.cell).then(cell => {
                     this.cell = cell
                     this.layout()
                 })
             }
-        }, this);
+        });
 
         ServerStateHandler.view("connectionStatus").addObserver((currentStatus, previousStatus) => {
```

Comment:
Might be able to use `observeKey` here too?

Reply:
Yes